### PR TITLE
Mohammad/frontend

### PR DESCRIPTION
**Pull request by mohammad-mallaee**

### CODE REVIEW

File: src/frontend/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+app.use(express.static('dist'))
+app.use(express.json())
+
+app.get("/evaluation", (req, res) => { res.json({ message: "Hello from server!" }) })
+
+app.get("/evaluation/teams", (req, res) => {
+    res.json(data.teams)
+})
+
+app.get("/evaluation/results/:id", (req, res) => {
```
Comment (AminMasoudi) [inline src/frontend/index.js:19]: the request url is like 
`/api/v1/evaluation/results/?team=7c90d362-c443-40da-b428-fab41656dd0f`

i can delete `/api/v1/` which is prefixed for all api's if you wish

File: src/frontend/src/utils/types.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    name: string
+}
+
+export type Result = {
+    id: string,
+    scenario: number,
+    state: "FAILED" | "DONE" | "PENDING"
+    score: number,
+    average_time: number
+}
```
Comment (AminMasoudi) [inline src/frontend/src/utils/types.ts:23]:
```suggestion
export type Result = {
    id: string,
    scenario: number,
    status: "FAILED" | "DONE" | "PENDING"
    score: number,
    average_time: number
}
```

this is an output of backend
```
    {
        "id": 1,
        "name": "qwerty",
        "weight": 2,
        "number_of_steps": 2
    }
```

however i can change it in backend

File: src/frontend/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+app.listen(PORT, () => {
+    console.log(`listening on port ${PORT}`)
+})
+
+app.use(express.static('dist'))
+app.use(express.json())
+
+app.get("/evaluation", (req, res) => { res.json({ message: "Hello from server!" }) })
+
+app.get("/evaluation/teams", (req, res) => {
```
Comment (AminMasoudi) [inline src/frontend/index.js:15]:
```suggestion
app.get("/api/v1/teams", (req, res) => {
```

File: src/frontend/src/lib.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { Team } from "./utils/types";
+
+export async function getTeams() {
+    try {
+        const res = await fetch(`/evaluation/teams`)
```
Comment (AminMasoudi) [inline src/frontend/src/lib.ts:5]:
```suggestion
        const res = await fetch(`/api/v1/teams`)
```

File: src/frontend/src/lib.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        return results
+    } catch (err: any) {
+        console.log(err)
+        throw new Error(err)
+    }
+}
+
+export async function getHistory(team: Team) {
+    try {
+        const res = await fetch(`/evaluation/history/${team.id}`)
```
Comment (AminMasoudi) [inline src/frontend/src/lib.ts:27]: # Not implement in backend yet 

```suggestion
        const res = await fetch(`/api/v1/team/${team.id}/history`)
```

**Review by AminMasoudi — CHANGES_REQUESTED** [review]
# Well Done

but there are a few conflicts i wish you to fix

tnx

File: src/frontend/index.js (hunk 1 above)
Comment (mohammad-mallaee) [reply src/frontend/index.js]: updated all routes to start with `/api/v1`

File: src/frontend/src/utils/types.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+export type Team = {
+    id: string,
+    name: string,
+    type: string,
+    total_score: number
+}
```
[the remark this answers is not in the document]
Comment (mohammad-mallaee) [reply src/frontend/src/utils/types.ts]: fixed

File: src/frontend/src/utils/types.ts (hunk 2 above)
Comment (mohammad-mallaee) [reply src/frontend/src/utils/types.ts]: changed state to status
output is for **Scenario** but this is **Result** type !!

File: src/frontend/index.js (hunk 3 above)
Comment (mohammad-mallaee) [reply src/frontend/index.js]: fixed

File: src/frontend/src/lib.ts (hunk 4 above)
Comment (mohammad-mallaee) [reply src/frontend/src/lib.ts]: fixed

File: src/frontend/src/lib.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        return teams
+    } catch (err: any) {
+        console.log(err)
+        throw new Error(err)
+    }
+}
+
+export async function getResults(team: Team) {
+    try {
+        const res = await fetch(`/evaluation/results/${team.id}`)
```
[the remark this answers is not in the document]
Comment (mohammad-mallaee) [reply src/frontend/src/lib.ts]: fixed

File: src/frontend/src/lib.ts (hunk 5 above)
Comment (mohammad-mallaee) [reply src/frontend/src/lib.ts]: fixed

File: src/frontend/src/utils/types.ts (hunk 2 above)
Comment (AminMasoudi) [reply src/frontend/src/utils/types.ts]:
```
 {
        "id": "13f2ebca-14a9-4096-b520-9228073dd3fe",
        "team_name": "qaz",
        "team": "7c90d362-c443-40da-b428-fab41656dd0f",
        "scenario": "http://127.0.0.1:8000/api/v1/evaluation/scenarios/1/",
        "status": "DONE",
        "average_time": 12,
        "score": 100
    }
```

File: src/frontend/src/lib.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        return results
+    } catch (err: any) {
+        console.log(err)
+        throw new Error(err)
+    }
+}
+
+export async function runTests(team: Team) {
+    try {
+        await fetch(`/api/v1/team/${team.id}/test`, {
```
Comment (AminMasoudi) [inline src/frontend/src/lib.ts:50]:
```suggestion
        await fetch(`/api/v1/teams/${team.id}/test`, {
```

File: src/frontend/src/lib.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        return results
+    } catch (err: any) {
+        console.log(err)
+        throw new Error(err)
+    }
+}
+
+export async function getHistory(team: Team) {
+    try {
+        const res = await fetch(`/api/v1/team/${team.id}/history`)
```
Comment (AminMasoudi) [inline src/frontend/src/lib.ts:27]:
```suggestion
        const res = await fetch(`/api/v1/teams/${team.id}/history`)
```

File: src/frontend/src/lib.ts (hunk 8 above)
Comment (mohammad-mallaee) [reply src/frontend/src/lib.ts]: fixed

File: src/frontend/src/lib.ts (hunk 9 above)
Comment (mohammad-mallaee) [reply src/frontend/src/lib.ts]: fixed

**Review by AminMasoudi — APPROVED** [review]
# Well Done

**Pull request merged**